### ISSUE TITLE
Error when using library in console command.

### ISSUE THREAD

**Issue by amirshawn** [spatie/laravel-medialibrary#851]
Hi,

I'm getting an error when running a command that accessing a model that uses the Media Library.  The error is:

Cannot use Spatie\MediaLibrary\Media as Media because the name is already in use {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\FatalErrorException(code: 64): Cannot use Spatie\\MediaLibrary\\Media as Media because the name is already in use at C:\\Cart\\app\\Product.php:11)

I am extending the media library and followed the directions to do so.  I have the new model reference in the config file.  Everything works fine when running code through the controllers but through the command I get that error.  Even without accessing media in the command, just looking up Product::all() I still get that error.  I think it has to do with me extending the model, possibly because my model is also called Media.  Just wondering if anyone's come across this problem?

**Comment by amirshawn** [comment]
I duplicated the Media model and named it CustomMedia.  Then I changed the medialibrary config file to reference the new model.  That stopped the error from occuring in the commands, but I think there's a bug if you use Media for the name of the extended model when running commands.

**Comment by freekmurze** [comment]
I'm thinking that somewhere you imported our `Media` (or used a `Media` alias) in a class where you already use another class named `Media`.

If you think the package has an error please submit a failing PR.

**Comment by amirshawn** [comment]
I'm confused.  Don't I have to import Media from your library?  This is what my Product model looks `like.

use Spatie\MediaLibrary\HasMedia\HasMediaTrait;
use Spatie\MediaLibrary\HasMedia\Interfaces\HasMediaConversions;
use Spatie\MediaLibrary\Media;

class Product extends Model implements Item, HasMediaConversions
{
    protected $guarded = ['id'];    
    use HasMediaTrait;

    public function registerMediaConversions(Media $media = null)
    {
        $this->addMediaConversion('icon_sm')
            ->width(50)
            ->height(50);
        $this->addMediaConversion('icon')
            ->width(100)
            ->height(100);
        $this->addMediaConversion('icon_lg')
            ->width(150)
            ->height(150);
        $this->addMediaConversion('thumb')
            ->width(250)
            ->height(250);
        $this->addMediaConversion('gallery')
            ->width(500)
            ->height(500);

    }

If I try importing my custom Media model then I get this error:

> Declaration of App\Product::registerMediaConversions(?App\Media $media = NULL) must be compatible with Spatie\MediaLibrary\HasMedia\Interfaces\HasMediaConversions::registerMediaConversions(?Spatie\MediaLibrary\Media $media = NULL)

I searched for Media:: to see if I'm calling that directly anywhere and it doesn't seem so. 

Thank you for your help!!

**Comment by freekmurze** [comment]
For some reason PHP think that your method signature is 

```php
App\Product::registerMediaConversions(?App\Media $media = NULL) 
```

It should be 
```
Spatie\MediaLibrary\HasMedia\Interfaces\HasMediaConversions::registerMediaConversions(?Spatie\MediaLibrary\Media $media = NULL)
```

In your code the namespace seems to be imported correctly.

Are you sure that class is being used?

Pretty sure this problem originates with your specific app and not this package.

**Comment by amirshawn** [comment]
Thank you for the response.  I'm not sure how to debug this.  I deleted everything in the command and ran Category::all();  and got the same error.  The category model uses the HasMediaConversions interface as well.
 I changed the name of my Media model  to CustomMedia and now everything is working fine.   So somehow the names are conflicting.   What confuses me is that I've had no problems running the same code through a controller but as a command it errors.  I'll just use the CustomMedia model.
 I just wanted to see if this has come up before and maybe there was a known solution.  I appreciate the help though and the great packages.  As soon as I finish this site, I will send you some postcardware :)
